src/CSnakes.SourceGeneration/PythonStaticGenerator.cs

```diff
@@ -58,26 +66,114 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             }
 
             // Parse the Python file
-            var result = PythonParser.TryParseFunctionDefinitions(code, out PythonFunctionDefinition[] functions, out GeneratorError[]? errors);
-
-            foreach (var error in errors)
+            try
             {
-                // Update text span
-                Location errorLocation = Location.Create(file.Path, TextSpan.FromBounds(0, 1), new LinePositionSpan(new LinePosition(error.StartLine, error.StartColumn), new LinePosition(error.EndLine, error.EndColumn)));
-                sourceContext.ReportDiagnostic(Diagnostic.Create(new DiagnosticDescriptor("PSG004", "PythonStaticGenerator", error.Message, "PythonStaticGenerator", DiagnosticSeverity.Error, true), errorLocation));
+                var result = PythonParser.TryParseFunctionDefinitions(code, out PythonFunctionDefinition[] functions, out GeneratorError[]? errors);
+
+                foreach (var error in errors)
+                {
+                    // Update text span
+                    Location errorLocation = Location.Create(file.Path, TextSpan.FromBounds(0, 1), new LinePositionSpan(new LinePosition(error.StartLine, error.StartColumn), new LinePosition(error.EndLine, error.EndColumn)));
+                    sourceContext.ReportDiagnostic(Diagnostic.Create(new DiagnosticDescriptor("PSG004", "PythonStaticGenerator", error.Message, "PythonStaticGenerator", DiagnosticSeverity.Warning, true), errorLocation));
+                }
+
+                if (result)
+                {
+                    var methods = ModuleReflection.MethodsFromFunctionDefinitions(functions).ToImmutableArray();
+                    string source = FormatClassFromMethods(@namespace, pascalFileName, methods, moduleAbsoluteName, functions, code, embedSourceSwitch);
+                    sourceContext.AddSource(generatedFileName, source);
+                    sourceContext.ReportDiagnostic(Diagnostic.Create(new DiagnosticDescriptor("PSG002", "PythonStaticGenerator", $"Generated {generatedFileName} from {file.Path}", "PythonStaticGenerator", DiagnosticSeverity.Info, true), Location.None));
+                }
+            } catch (Exception ex) {
```

Comment:
Generally it is discourages to catch any kind of exception. Can we be more specific? This seems to be a dramatic change of policy and wondering what prompted this.

Reply:
Exceptions thrown by a source generator are captured in the build logs but they're hard to find, log, and debug. Also they don't contain any details on which source file caused the exception. You'll see when compiling the stdlib integration tests it still has a few exceptions because there are unsupported syntax features.
 
IF any user gets an error during build because of a bug in the CSnakes source generator we get a better error message as well which they can share in the bug reports.